doc/source/rllib-concepts.rst

```diff
@@ -581,12 +581,141 @@ Here is an example of creating a set of rollout workers and using them gather ex
+
+ - ``round_robin``: Alternate taking items from each input dataflow. This ensures a fixed ratio of computations between, e.g., experience generation and experience replay. The ratio can be adjusted by setting ``round_robin_weights``.
+ - ``async``: Execute each input dataflow as fast as possible without blocking. You might want to use this when, e.g., you want replay to proceed as fast as possible irregardless of how fast experiences are being generated.
+
+**Metric ops** (`metric_ops.py <https://github.com/ray-project/ray/blob/master/rllib/execution/metric_ops.py>`__): Finally, we provide a ``StandardMetricsReporting`` operator that collects training metrics from the rollout workers in a unified fashion, and returns a stream of training result dicts. Execution plans should always end with this operator. This metrics op also reports various internal performance metrics stored by other operators in the shared metrics context accessible via ``_get_shared_metrics()``.
+
+Example: Asynchrony
+~~~~~~~~~~~~~~~~~~~~
+
+Suppose we wanted to make the above A2C example asynchronous (i.e., A3C). We would switch the synchronous ``ParallelRollouts`` operator with ``AsyncGradients``, and use ``ApplyGradients`` to apply gradient updates as fast as they are collected:
+
+.. code-block:: python
+
+    def execution_plan(workers: List[RolloutWorker], config: TrainerConfigDict):
+        # type: LocalIterator[(ModelGradients, int)]
+        grads = AsyncGradients(workers)
+
+        # type: LocalIterator[_]
+        train_op = grads.for_each(ApplyGradients(workers, update_all=False))
+
+        # type: LocalIterator[ResultDict]
+        return StandardMetricsReporting(train_op, workers, config)
+
+See also the `actual A3C implementation <https://github.com/ray-project/ray/blob/master/rllib/agents/a3c/a3c.py>`__.
+
+Example: Replay
+~~~~~~~~~~~~~~~
+
+Let's try adding a replay buffer to A2C. This can be done as follows by inserting store / replay ops and using ``Concurrently`` to compose them together:
+
+.. code-block:: python
+
+    def execution_plan(workers: List[RolloutWorker], config: TrainerConfigDict):
+        # Construct a replay buffer.
+        replay_buffer = LocalReplayBuffer(...)
+
+        # type: LocalIterator[_]
+        store_op = ParallelRollouts(workers, mode="bulk_sync") \
+            .for_each(StoreToReplayBuffer(local_buffer=replay_buffer))
+
+        # type: LocalIterator[(SampleBatchType, List[LearnerStatsDict])]
+        replay_op = Replay(local_buffer=replay_buffer) \
+            .for_each(TrainOneStep(workers))
+
+        # type: LocalIterator[(SampleBatchType, List[LearnerStatsDict])]
+        train_op = Concurrently(
+            [store_op, replay_op], mode="round_robin", output_indexes=[1])
```

Comment:
Can you explain output_indices again here (I know you mentioned it above in the explanation of Concurrently, but still not very clear, what it actually does to the resulting `train_op`)?